clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -482,7 +484,7 @@ public StoreIngestionTask(
     this.ingestionNotificationDispatcher =
         new IngestionNotificationDispatcher(notifiers, kafkaVersionTopic, isCurrentVersion);
     this.missingSOPCheckExecutor.execute(() -> waitForStateVersion(kafkaVersionTopic));
-    this.chunkAssembler = new ChunkAssembler(storeName);
+    this.chunkAssembler = new InMemoryChunkAssembler(new InMemoryStorageEngine(storeName));
```

Comment:
This instance is mainly for DVRT without an local state, right?
I think we are considering keeping a local state in some cases, so I wondered whether we need to construct an `InMemoryChunkAssembler` all the time or not.
Or shall the `ChunkAssembler` be smart enough to decide whether to use an InMemoryStorageEngine or on-disk RocksDB internally or not.

Reply:
Maybe I'm missing some context about DVRT but are you worried about the cost where we instantiate a `InMemoryChunkAssembler` but never actually use it?

Reply:
Two parts:
1. Unnecessary instance creation, and overhead is minimal, but mainly for best practice of coding and if we always create many unnecessary objects, eventually, they can accumulate to incur a bigger cost.
2. Whether InMemoryStorageEngine should be used when DVRT is also trying to keep a local RocksDB snapshot, and I assume when DVRT does keep a local snapshot, we don't have to use InMemoryStorageEngine since all chunks are already in local disk.